modules/pricer/src/main/java/com/opengamma/strata/pricer/ZeroRateDiscountFactors.java

```diff
@@ -168,11 +170,17 @@ public double relativeYearFraction(LocalDate date) {
   @Override
   public double discountFactor(double yearFraction) {
     // convert zero rate to discount factor
+    if (yearFraction <= EFFECTIVE_ZERO) {
```

Comment:
If `discountFactor` is modified, `zeroRate` needs to be also modified for consistency. By the current change, we have discountFactor = 1, but nonzero yearFraction*zeroRate, for negative year fractions.